cmake/kconfig.cmake

```diff
@@ -0,0 +1,70 @@
+file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/include/generated)
+# Folders needed for conf/mconf files (kconfig has no method of redirecting all output files).
+# conf/mconf needs to be run from a different directory because of: ZEP-1963
+file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/kconfig/include/generated)
+file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/kconfig/include/config)
+
+set(BOARD_DEFCONFIG ${PROJECT_SOURCE_DIR}/boards/${ARCH}/${BOARD}/${BOARD}_defconfig)
+set(APP_CONFIG      ${APPLICATION_SOURCE_DIR}/prj.conf)
```

Comment:
This should respect the CONF_FILE variable instead (or provide a similar user-controllable means for CMake), and handle the case that it's a list of files to merge instead of assuming it's just one file, to match current behavior.